qubekit/fragmentation/WBOFragmenter.py

```diff
@@ -0,0 +1,228 @@
+import itertools
+from typing import List, TYPE_CHECKING
+from typing_extensions import Literal
+
+from pydantic import Field
+from qcelemental.util import which_import
+from openff.toolkit.topology import Molecule as OFFMolecule
+from rdkit import Chem
+
+from qubekit.molecules import Ligand
+from qubekit.utils.datastructures import StageBase
+
+if TYPE_CHECKING:
+    from openff.toolkit.topology import Molecule
+    from openff.fragmenter.fragment import (
+        FragmentationResult,
+    )
+
+
+class WBOFragmenter(StageBase):
+
+    type: Literal["WBOFragmenter"] = "WBOFragmenter"
+
+    rotatable_smirks: List[str] = Field(
+        ["[!#1]~[!$(*#*)&!D1:1]-,=;!@[!$(*#*)&!D1:2]~[!#1]"],
+        description="The rotatable smirks to use for fragmentation",
+    )
+
+    @classmethod
+    def is_available(cls) -> bool:
+        fragmenter = which_import(
+            "openff.fragmenter",
+            return_bool=True,
+            raise_error=True,
+            raise_msg="Please install via `conda install -c conda-forge openff-fragmenter`.",
+        )
+
+        rdkit = which_import(
+            "rdkit",
+            return_bool=True,
+            raise_error=True,
+            raise_msg="Please install via `conda install -c conda-forge rdkit`.",
+        )
+
+        return fragmenter and rdkit
+
+    def _prepare_fragment_input(self, molecule: Ligand):
+        """
+        TODO
+        """
+        # convert to OFF molecule
+        mapped_smiles = molecule.to_smiles(mapped=True)
+        off_molecule: OFFMolecule = OFFMolecule.from_mapped_smiles(mapped_smiles)
+
+        # remember the atom names
+        for off_atom, lig_atom in zip(off_molecule.atoms, molecule.atoms):
+            assert off_atom.atomic_number == lig_atom.atomic_number
+            off_atom.name = lig_atom.atom_name
+
+        # convert to the canonical order which will be used by the fragmenter
+        return off_molecule.canonical_order_atoms()
```

Comment:
I think we can drop all of this as we are just going to use the canonical ordered molecule from the result of the fragmentation.